### PR TITLE
StarknetDomainHost

### PR DESCRIPTION
**Pull request by maciejka**

### CODE REVIEW

File: src/tests/domains/StarknetDomainHost.t.sol [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+        assertEq(host.l2Dai(), L2_DAI);
+
+        assertEq(vat.can(address(host), address(daiJoin)), 1);
+        assertEq(dai.allowance(address(host), address(daiJoin)), type(uint256).max);
+        assertEq(dai.allowance(address(host), address(router)), type(uint256).max);
+        assertEq(host.wards(address(this)), 1);
+        assertEq(host.live(), 1);
+    }
+
+    function testDepost() public {
```
Comment (EvolveArt) [inline src/tests/domains/StarknetDomainHost.t.sol]: might be `testDeposit`

File: src/tests/domains/StarknetDomainHost.t.sol [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+    function testUnauthorizedSetMaxDeposit() public {
+        host.deny(address(this));
+        vm.expectRevert('DomainHost/not-authorized');
+        host.setMaxDeposit(12 ether);
+    }
+
+}
+
+function split(uint256 value) pure returns (uint256, uint256) {
```
Comment (EvolveArt) [inline src/tests/domains/StarknetDomainHost.t.sol:182]: can't this be imported from `StarknetDomainHost` ?

**Review by EvolveArt — COMMENTED** [review]
Shouldn't we also test withdrawal + settlement logic there ? Or that'd just be in the integration tests ?

File: src/tests/domains/StarknetDomainHost.t.sol (hunk 2 above)
Comment (maciejka) [reply src/tests/domains/StarknetDomainHost.t.sol:182]: Not sure if it deduplication is worth the resulting extension of StarknetDomainHost public interface.

**Comment by maciejka** [comment]
> Shouldn't we also test withdrawal + settlement logic there ? Or that'd just be in the integration tests ?

I want to unit test here what is not already tested by other unittests. So basic withdrawal tests will be added later.

File: src/tests/domains/StarknetDomainHost.t.sol (hunk 2 above)
Comment (EvolveArt) [reply src/tests/domains/StarknetDomainHost.t.sol:182]: Right, but we might also need it in integration tests

**Comment by maciejka** [comment]
#14 is more up to date

**Pull request closed without merging**